nncf/nncf_network.py

```diff
@@ -747,3 +759,26 @@ def get_MACs_in_model(self):
 
     def get_input_infos(self) -> List[ModelInputInfo]:
         return deepcopy(self.input_infos)
+
+    @staticmethod
+    def collect_train_only_module_scopes(model, graph_builder) -> List[str]:
```

Comment:
You are iterating over nodes - these do not necessarily represent entire module calls. Consider the situation:
```
*some module's forward*
x = self.op_1(x)
if self.training:
    y = self.train_only_op(x)
x = self.op_2(x)
```
Your code, it seems, will effectively remove `op_1` and `op_2` calls from the compression consideration as well, which may be incorrect.

Reply:
Corrected logic by processing operation execution context.  A module is wrapped, if they're at least a single operation or a module called in the eval mode inside it.